openshift/system/Dockerfile

```diff
@@ -1,82 +1,75 @@
+ADD . ./
+ADD openshift/system/config/* ./config/
+RUN dnf -y module enable ruby:${RUBY_VERSION} nodejs:16 mysql:8.0 \
+    && dnf install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs shared-mime-info make automake gcc gcc-c++ redhat-rpm-config postgresql rubygem-irb rubygem-rdoc ruby-devel nodejs libpq-devel mysql-devel gd-devel git 'dnf-command(download)' \
+    && gem install bundler --version 2.2.25 \
+    && bundle config set --local without development:test:licenses \
+    && bundle config set --local deployment true \
+    && bundle install --jobs $(grep -c processor /proc/cpuinfo) --retry=5 \
+    && npm install -g yarn \
+    && yarn install:safe --no-progress \
+    && SECRET_KEY_BASE=rails/32947 bundle exec rake assets:precompile tmp:clear \
+    && rm -rf tmp/cache /usr/local/share/gems/cache /usr/local/share/gems/doc
 
-ENV RUBY_VERSION="${RUBY_MAJOR_VERSION}.${RUBY_MINOR_VERSION}" \
-    RUBY_SCL_NAME_VERSION="${RUBY_MAJOR_VERSION}${RUBY_MINOR_VERSION}"
+# can just install memkind once it is fixed, see
+# https://issues.redhat.com/browse/RHEL-14497
+RUN cd /tmp \
+    && dnf download --source memkind \
+    && rpm2cpio memkind-*.src.rpm | cpio -idmv "memkind-*.tar.gz" \
+    && tar xvfz memkind-*.tar.gz \
+    && cd memkind-*/jemalloc/ \
+    && ./autogen.sh && ./configure --libdir=/usr/local/lib64/ && make
 
-ENV TZ=:/etc/localtime \
-    BUNDLE_GEMFILE=Gemfile \
-    VARNISH_SCL=rh-varnish6 \
-    NODEJS_SCL=rh-nodejs14 \
-    RUBY_SCL="rh-ruby${RUBY_SCL_NAME_VERSION}" \
-    GIT_SCL=rh-git227
 
-WORKDIR /opt/system
+FROM quay.io/centos/centos:stream8 AS base
 
-RUN yum-config-manager --save --setopt=cbs.centos.org_repos_sclo7-$RUBY_SCL-rh-candidate_x86_64_os_.skip_if_unavailable=true \
-    && rpm -Uvh https://download.postgresql.org/pub/repos/yum/reporpms/EL-7-x86_64/pgdg-redhat-repo-latest.noarch.rpm \
-    && yum -y update \
-    && yum remove -y postgresql \
-    && yum install -y epel-release \
-    && yum -y install centos-release-scl-rh \
-              gd-devel \
-              unixODBC-devel \
-              mysql \
-              liberation-sans-fonts \
-              llvm5.0-devel \
-              postgresql13 postgresql13-devel postgresql13-libs \
-              file \
-              $NODEJS_SCL \
-              $VARNISH_SCL-jemalloc \
-              $GIT_SCL \
-    && yum -y clean all \
-    && scl enable $VARNISH_SCL -- printf '$LIBRARY_PATH' | cut -d: -f1 > /etc/ld.so.conf.d/jemalloc.conf \
-    && ldconfig && ldconfig -p | grep jemalloc
+USER root
 
-# We don't want to redo the bundle install step every time a file has changed:
-# copying only the gemspec files and copying all the other files after the build
-ADD lib/developer_portal/*.gemspec lib/developer_portal/
-ADD vendor/active-docs/*.gemspec vendor/active-docs/
-ADD Gemfile* ./
-ADD Rakefile ./
+ENV RUBY_MAJOR_VERSION=2 \
+    RUBY_MINOR_VERSION=7 \
+    RAILS_ENV=production \
+    SAFETY_ASSURED=1 \
```

Comment:
I still think `SAFETY_ASSURED` is not needed. DB is initialized from schema, not migrations. And we are in production anyway, where `strong_migrations` doesn't intervene.

Reply:
no worries, it's gone locally, not pushed yet, unless I mixed things up, I'll double check

Reply:
Update, it is not gone because it was there previously and I don't want to change this in this PR. If you like to remove it, lets do in another PR but somebody has to test whether it may cause troubles by certain DB migrations.